### DISCUSSION TITLE
Migrating from JS to TS partially throws TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts"

### DISCUSSION THREAD

**Discussion by unknowntester404** [webdriverio/webdriverio#12662] [Q&A] [answered]
Hello,

I am trying to migrated from JS to TS incrementally. So I've started with two files under constants folder called variables.ts and data.ts
This two files are then used in wdio.conf.js file via import {} from "../constants/variables.ts" for example

However when running yarn test then @wdio/config:ConfigParser: Failed loading configuration file: file:///Users/****/Projects/**/configs/wdio.conf.js: Unknown file extension ".ts" for /Users/****/Projects/**/constants/variables.ts
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /Users/****/Projects/**/constants/variables.ts

my tsconfig.json looks like this
```
{
  "compilerOptions": {
    "target": "ES6",
    "module": "ESNext",
    "moduleResolution": "NodeNext",
    "esModuleInterop": true,
    "types": [
      "node",
      "@wdio/globals/types",
      "@wdio/mocha-framework",
      "@wdio/appium-service"
    ],
    "allowImportingTsExtensions": true,
    "noEmit": true,
    "forceConsistentCasingInFileNames": true
  },
  "include": ["/constants/*.ts"],
  "exclude": [
    "/node_modules",
    "/pages/",
    "/test/",
  ]
}
```
and my package json has type:"module" inside it.

How can make yarn tests work again? Thank

**Comment by christian-bromann** [comment]
> This two files are then used in wdio.conf.js file via import {} from "../constants/variables.ts" for example

Even though you are importing a `.ts` file you have to use the `.js` file extension. Can you try rewriting your import to:

```ts
import {} from "../constants/variables.js"
```
?

**Reply by unknowntester404** [reply]
This throws
```
import {VAR1, VAR2, VAR3} from '../constants/variables.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

**Reply by christian-bromann** [reply]
This error means that your project uses CJS which requires you to import via `const mod = require("mod")` instead of `import mod from 'mod'`. Make sure you set `"module": "commonjs",` in your `tsconfig.json`.

**Reply by jurjandrei** [reply, marked as answer]
Managed to solve this by changing to "target": "ES2016"
